### DISCUSSION TITLE
怎么指定一个项目只有release版本，而不影响其他项目？

### DISCUSSION THREAD

**Discussion by Domain** [xmake-io/xmake#5619] [Q&A]
我这边编译的流程是这样：
编译工具 --> 执行工具 --> 编译其他项目
工具会生成一些代码，所以必须先执行。平时开发项目一般都是编译debug版本，但是工具我不希望使用debug版本，因为执行起来太慢，我希望无论其他项目是release还是debug，工具都按照release编译，现在项目配置大概如下：
```lua
target("builder")
    set_kind("binary")
    add_files("builder/**.cpp")

rule("xml")
    set_extensions(".xml")
    on_build_files(function (target, sourcebatch)
        local builder = target:dep("builder"):targetfile()
        local xmls = table.concat(sourcebatch.sourcefiles, ",")
        os.execv(builder, { "-f", xmls })
    end)
    on_link(function (target)
    end)
rule_end()

target("xml")
    set_kind("binary")
    add_rules("xml")
    add_deps("builder")
    add_files("xml/**.xml")

target("app")
    set_kind("binary")
    add_files("app/**.cpp")
    add_deps("xml")
```
这种配置编译debug版本app时，builder也是debug版本，执行起来太慢。有什么办法让builder永远是release版本，除非设置了某个参数。

**Comment by waruqi** [comment]
参考 release rules 的设置，对 builder target 单独设置上不就好了么。。

```lua
target("builder")
    set_kind("binary")
    add_files("builder/**.cpp")
    set_optimize("fastest")
    set_strip("all")
    set_symbols("hidden")
```

https://github.com/xmake-io/xmake/blob/0c42d253e0b9cb39ebdd4cdd22516f80cea7dc0e/xmake/rules/mode/xmake.lua#L57

**Reply by Domain** [reply]
我试了一下，不行，是不是`xmake f -m debug`会覆盖这个？

**Reply by waruqi** [reply]
不会，不可能不行，自己 check 下 -vD 的 flags ，不要看到进度里面显示 debug ，就一定是 debug

**Reply by Domain** [reply]
我确实没有认真确认过，只是我用release模式编译出来的exe大小和这种设置方式编译出来的不一样，release模式小很多。我想了一下，可能是builder依赖的库还是使用debug版本导致的
